ministryofjustice/peoplefinder
# CT-3137 Add new cronjob for image deletion

## Description
Add new cronjob for image deletion, as with other products, this deletes tagged images so that:
10 latest master images are kept and
non-master images in the last 14 days are kept
All other tagged images are deleted
Untagged images may not be deleted - but there are not many of these anyway - as the script does give the figures. This may be another iteration for all products if it becomes important to delete few untagged images.

## Self-review checklist


### Screenshots
Making sure no non-master images newer than 14 days deleted:
![image](https://user-images.githubusercontent.com/22935203/98127204-568a9e80-1eae-11eb-953e-de5e9328f704.png)

Make sure that all master images to delete are master images (Yes, 10 found, all say master)

Making sure no more than 10 master images are kept
(Sublime search returned 10 results)

Make sure that the 10 master images are kept (master_images_to_keep)
Yes - 10 searched in sublime, and all say master 


2 Anomaly images (untagged):
![image](https://user-images.githubusercontent.com/22935203/98128827-48d61880-1eb0-11eb-894a-8a08dd7f576a.png)

Makesure correct images are deleted:
![image](https://user-images.githubusercontent.com/22935203/98128034-4cb56b00-1eaf-11eb-90b4-2eaa56fb0fbd.png)

Makesure that all remaining images contain 10 master images:
![image](https://user-images.githubusercontent.com/22935203/98128587-fd236f00-1eaf-11eb-9027-316f0b5a2c4b.png)

Make sure that all remaining images only contain non-master images that are in the last 14 days:
(older than 14 - none)
![image](https://user-images.githubusercontent.com/22935203/98129553-2a245180-1eb1-11eb-9125-809874c44acc.png)
(newer than 14 - 7)


### Related JIRA tickets
https://dsdmoj.atlassian.net/browse/CT-3137

### Deployment
deployment file changed, so deployment should be tested, and if production, Cronjob should be applied

### Manual testing instructions
Check screenshots for script. Also once applied to prod, check if cronjob is there by doing:
kubectl get cronjobs -n peoplefinderj-production
